### PR TITLE
Remove Honeywell provider

### PR DESCRIPTION
**Pull request by vxfield**
### Context
In November 2021, Honeywell became Quantinuum: Press release: [Quantinuum is a result of the combination of two global leaders in quantum computing: Honeywell Quantum Solutions and Cambridge Quantum](https://www.quantinuum.com/pressrelease/introducing-quantinuum)

In February/March 2022, Azure Quantum started supporting the Quantinuum provider and targets.

### This PR
This PR removes the Honewell provider, targets and tests.

### CODE REVIEW

**Comment by ricardo-espinoza** [comment]
@vxfield , I think this will require an update to the recordings, and at the same time eliminating those from Honeywell tests.

**Comment by vxfield** [comment]
@ricardo-espinoza, yes, work-in-progress :-)

**Comment by vxfield** [comment]
@ricardo-espinoza, ready for final review now

File: azure-quantum/azure/quantum/qiskit/backends/quantinuum.py
Comment (ricardo-espinoza) [inline azure-quantum/azure/quantum/qiskit/backends/quantinuum.py:56]: Ha! Thanks for catching that. In any case, we should remove the "(formerly Honeywell)" entirely here since it's gone from other parts of the code. No need to block the PR for this, so I'll signoff anyway.

**Review by ricardo-espinoza — APPROVED** [review]

File: azure-quantum/azure/quantum/qiskit/backends/quantinuum.py
Comment (vxfield) [reply azure-quantum/azure/quantum/qiskit/backends/quantinuum.py:56]: Ok. I'll let it there then and we can remove at the next PR.

**Pull request merged**
